### PR TITLE
HP-1861: Leasing add

### PR DESCRIPTION
**Pull request by bladeroot**
## Summary by CodeRabbit

- **New Features**
	- Introduced a `LeasingWasStarted` event to signify the start of a leasing period.
	- Added functionality to handle the beginning and ending of leasing charges.
- **Enhancements**
	- Improved leasing functionality with more descriptive method names and additional checks for the first month of leasing.
- **Tests**
	- Updated the Leasing feature tests to include scenarios with the `LeasingWasStarted` event.

### CODE REVIEW

File: src/sale/Sale.php [hunk 1]
```diff
@@ -125,6 +129,11 @@ public function setId($id)
         $this->id = $id;
     }
 
+    public function getData()
```
Comment (SilverFire) [inline src/sale/Sale.php:132]: Return data type

**Review by SilverFire — CHANGES_REQUESTED** [review]
How the PR name "Leasing add" relates to what's happening inside?

**Review by SilverFire — CHANGES_REQUESTED** [review]
I don't see how you use these changes in the application code. Could you explain you why need it?

File: src/sale/SaleInterface.php [hunk 2]
```diff
@@ -61,6 +61,11 @@ public function getTime();
 
     public function getCloseTime(): ?DateTimeImmutable;
 
+    /**
+     * @return string|json|array|null
+     */
+    public function getData();
```
[the remark this answers is not in the document]
Comment (bladeroot) [reply src/sale/SaleInterface.php]: +

File: src/sale/Sale.php (hunk 1 above)
Comment (bladeroot) [reply src/sale/Sale.php:132]: in SaleInterface

**Review by SilverFire — APPROVED** [review]

**Pull request merged**